app/models/employee.rb

```diff
@@ -19,19 +19,36 @@ class Employee < ApplicationRecord
   belongs_to :company_branch
   has_many :attendances, dependent: :destroy
 
+  validates :name, :email, :position, :employee_number, :private_number, presence: true
+  validates :email, uniqueness: true
+  validates :private_number, uniqueness: true
+  validates :employee_number, uniqueness: true
+
   scope :actives, -> { where(is_active: true) }
 
+  def self.search(search_name)
+    if search_name
+      self.where('name LIKE ?', "%#{search_name}%")
+    else
+      self.order(:name)
+    end
+  end
+
   def absence_by_month
-    attendances.where.not(check_in: nil).where('check_in BETWEEN ? AND ?', DateTime.now.beginning_of_month,
-                                               DateTime.now.end_of_month).count
+    DateTime.now.day - attendances.where(
+      'check_in BETWEEN ? AND ?',
+      DateTime.now.beginning_of_month,
+      DateTime.now.end_of_month
+    ).count
```

Comment:
take care of the indentation of the code

Reply:
ok